pkg/gadgets/trace/capabilities/types/types.go

```diff
@@ -34,16 +34,19 @@ const (
 type Event struct {
 	eventtypes.Event
 
-	Pid       uint32 `json:"pid,omitempty" column:"pid,template:pid"`
-	Comm      string `json:"comm,omitempty" column:"comm,template:comm"`
-	Syscall   string `json:"syscall,omitempty" column:"syscall,template:syscall"`
-	UID       uint32 `json:"uid,omitempty" column:"uid,minWidth:6"`
-	Cap       int    `json:"cap,omitempty" column:"cap,width:3,fixed"`
-	CapName   string `json:"capName,omitempty" column:"capName,width:18,fixed"`
-	Audit     int    `json:"audit,omitempty" column:"audit,minWidth:5"`
-	Verdict   string `json:"verdict,omitempty" column:"verdict,width:7,fixed"`
-	InsetID   *bool  `json:"insetid,omitempty" column:"insetid,width:7,fixed,hide"`
-	MountNsID uint64 `json:"mountnsid,omitempty" column:"mntns,template:ns"`
+	Pid           uint32 `json:"pid,omitempty" column:"pid,template:pid"`
+	Comm          string `json:"comm,omitempty" column:"comm,template:comm"`
+	Syscall       string `json:"syscall,omitempty" column:"syscall,template:syscall"`
+	UID           uint32 `json:"uid,omitempty" column:"uid,minWidth:6"`
+	Cap           int    `json:"cap,omitempty" column:"cap,width:3,fixed"`
+	CapName       string `json:"capName,omitempty" column:"capName,width:18,fixed"`
+	Audit         int    `json:"audit,omitempty" column:"audit,minWidth:5"`
+	Verdict       string `json:"verdict,omitempty" column:"verdict,width:7,fixed"`
+	InsetID       *bool  `json:"insetid,omitempty" column:"insetid,width:7,fixed,hide"`
+	MountNsID     uint64 `json:"mountnsid,omitempty" column:"mntns,template:ns"`
+	TargetUserNs  uint64 `json:"targetuserns,omitempty" column:"targetuserns,template:ns"`
+	CurrentUserNs uint64 `json:"currentuserns,omitempty" column:"currentuserns,template:ns"`
+	CapEffective  string `json:"capeffective,omitempty" column:"capeffective,hide"`
```

Comment:
Using a hexadecimal string is nice because it can be copy/pasted into `capsh --decode=`  or even read manually by seeing bit 0 to bit 40 set (41 bits total) set:

```
$ cat /proc/sys/kernel/cap_last_cap
40
```

And it would be difficult for a custom extractor to print the very long string like the following:
> cap_chown,cap_dac_override,cap_dac_read_search,cap_fowner,cap_fsetid,cap_kill,cap_setgid,cap_setuid,cap_setpcap,cap_linux_immutable,cap_net_bind_service,cap_net_broadcast,cap_net_admin,cap_net_raw,cap_ipc_lock,cap_ipc_owner,cap_sys_module,cap_sys_rawio,cap_sys_chroot,cap_sys_ptrace,cap_sys_pacct,cap_sys_admin,cap_sys_boot,cap_sys_nice,cap_sys_resource,cap_sys_time,cap_sys_tty_config,cap_mknod,cap_lease,cap_audit_write,cap_audit_control,cap_setfcap,cap_mac_override,cap_mac_admin,cap_syslog,cap_wake_alarm,cap_block_suspend,cap_audit_read

Reply:
What I meant is to transport the data using an unsigned integer and then print to the user as hexadecimal. (It's actually not that important, but I feel bad when we use strings to transport something that can be represented with an integer).

Reply:
I made it in two columns:
* caps: as uint64, with a column extractor to display the hexadecimal
* capsnames, with a comma-separated list, following the same model as the existing columns cap and capname, so other GUIs don't have to decode it. It's hidden by default, so the long list is not bothering.